### ISSUE TITLE
Revamp test framework

### ISSUE THREAD

**Issue by cmungall** [linkml/linkml#335]
- [ ] Move from unittest to pytest (in progress)
- [x] Use uniform compliance tests across generators

**Comment by WolfgangFahl** [comment]
Could you be a bit more specific about this? What would be the definition of done?

**Comment by cmungall** [comment]
When all tests use pytest

**Comment by vladistan** [comment]
> When all tests use pytest

I noticed many (all?) unit tests in `linkml-runtime` still use `UnitTest` style of tests.  Should they be converted to `pytest` as well?

**Comment by sierra-moxon** [comment]
Yes, that is the direction; we would welcome PRs in this direction. 😁
